src/ts/ConvivaAnalytics.ts

```diff
@@ -657,79 +698,127 @@ export class ConvivaAnalytics {
     this.convivaVideoAnalytics.reportAdBreakStarted(
       Conviva.Constants.AdType.CLIENT_SIDE,
       Conviva.Constants.AdPlayer.SEPARATE,
     );
   };
 
-  private onAdBreakFinished = (event: AdBreakEvent | ErrorEvent) => {
-    this.debugLog('[ ConvivaAnalytics ] adbreak finished', event);
-    this.isAd = false;
+  private onAdStarted = (event: AdEvent) => {
+    this.debugLog('[ Player Event ] ad started', event);
 
-    if (!this.isSessionActive()) {
-      // Don't report without a valid session (e.g., in case of a pre-roll, or post-roll ad)
-      return;
+    const adInfo = AdHelper.extractConvivaAdInfo(this.player, this.lastAdBreakEvent, event);
+    // It's already in kbps
+    const bitrateKbps = event.ad.data?.bitrate;
+
+    this.debugLog('[ ConvivaAdAnalytics ] report ad started', {
+      event,
+      adInfo,
+    });
+    this.convivaAdAnalytics.reportAdStarted(adInfo);
+
+    this.debugLog('[ ConvivaAdAnalytics ] report playing ad playback state');
+    this.convivaAdAnalytics.reportAdMetric(Conviva.Constants.Playback.PLAYER_STATE, Conviva.Constants.PlayerState.PLAYING);
+
+    if (bitrateKbps) {
+      this.debugLog('[ ConvivaAdAnalytics ] report ad bitrate', bitrateKbps);
+      this.convivaAdAnalytics.reportAdMetric(Conviva.Constants.Playback.BITRATE, bitrateKbps);
     }
+  }
+
+  // Fires before `onAdBreakFinished`
+  private onAdFinished = (event: AdEvent) => {
+    this.debugLog('[ Player Event ] ad finished', event);
+
+    this.debugLog('[ ConvivaAdAnalytics ] report ad ended', {
+      event,
+    });
+    this.convivaAdAnalytics.reportAdEnded();
+  }
 
+  // Fires before `onAdBreakFinished`
+  private onAdSkipped = (event: AdEvent) => {
+    this.debugLog('[ Player Event ] ad skipped', event);
+
+    this.debugLog('[ ConvivaAdAnalytics ] report ad skipped', event);
+    this.convivaAdAnalytics.reportAdSkipped();
+
+    this.onCustomEvent(event);
+  };
+
+  private onAdBreakFinished = (event: AdBreakEvent | ErrorEvent) => {
+    this.debugLog('[ Player Event ] adbreak finished', event);
+    this.isAdBreak = false;
+
+    this.debugLog('[ ConvivaAnalytics ] report ad break ended', event);
     this.convivaVideoAnalytics.reportAdBreakEnded();
+
+    this.debugLog('[ ConvivaAnalytics ] report playing playback state');
     this.convivaVideoAnalytics.reportPlaybackMetric(
       Conviva.Constants.Playback.PLAYER_STATE,
       Conviva.Constants.PlayerState.PLAYING,
     );
   };
 
-  private onAdSkipped = (event: AdEvent) => {
-    this.onCustomEvent(event);
-  };
+  private onAdError = (event: ErrorEvent) => {
+    this.debugLog('[ Player Event ] ad error', event);
+
+    const formattedError = AdHelper.formatAdErrorEvent(event);
+
+    this.debugLog('[ ConvivaAdAnalytics ] report ad error', {
+      event,
+      formattedError,
+    });
+    this.convivaAdAnalytics.reportAdError(formattedError, Conviva.Constants.ErrorSeverity.WARNING);
 
-  private onAdError = (event: AdEvent) => {
     this.onCustomEvent(event);
```

Comment:
Same as above. Do we still need to report the error as custom event now we have the AdAnalytics? 🤔

Reply:
As I mentioned, I would keep it at least for some time.